hamocc/mo_carchm.F90

```diff
@@ -0,0 +1,880 @@
+    real,    intent(in)    :: Kf    ! equilibrium constant Kf  = [H][F]/[HF].
+    real,    intent(in)    :: Ksi   ! equilibrium constant Ksi = [H][SiO(OH)3]/[Si(OH)4].
+    real,    intent(in)    :: K1p   ! equilibrium constant K1p = [H][H2PO4]/[H3PO4].
+    real,    intent(in)    :: K2p   ! equilibrium constant K2p = [H][HPO4]/[H2PO4].
+    real,    intent(in)    :: K3p   ! equilibrium constant K3p = [H][PO4]/[HPO4].
+    real,    intent(inout) :: ah1   ! hydrogen ion concentration.
+    real,    intent(out)   :: ac    ! carbonate alkalinity.
+    integer, intent(in)    :: niter ! maximum number of iteration
+
+    ! Parameters to set accuracy of iteration
+    real,  parameter     :: eps=5.e-5
+
+    ! Local variables
+    integer :: jit
+    real    :: s,scl,borat,sti,ft
+    real    :: hso4,hf,hsi,hpo4,ab,aw,ah2o,ah2,erel
+
+    ! Calculate concentrations for borate, sulfate, and fluoride; see Dickson, A.G.,
+    ! Sabine, C.L. and Christian, J.R. (Eds.) 2007. Guide to best practices
+    ! for ocean CO2 measurements. PICES Special Publication 3, chapter 5 p. 10
+    s = MAX(25.,saln)
+    scl = s * salchl
+    borat = bor1 * scl * bor2      ! Uppstrom (1974)
+    sti = 0.14 * scl / 96.062      ! Morris & Riley (1966)
+    ft = 0.000067 * scl / 18.9984  ! Riley (1965)
+
+    iflag: do jit = 1,niter
+      hso4 = sti / ( 1. + Ks1 / ( ah1 / ( 1. + sti / Ks1 ) ) )
+      hf   = 1. / ( 1. + Kf / ah1 )
+      hsi  = 1./ ( 1. + ah1 / Ksi )
+      hpo4 = ( K1p * K2p * ( ah1 + 2. * K3p ) - ah1**3 ) /    &
+           ( ah1**3 + K1p * ah1**2 + K1p * K2p * ah1 + K1p * K2p * K3p )
+      ab   = borat / ( 1. + ah1 / Kb )
+      aw   = Kw / ah1 - ah1 / ( 1. + sti / Ks1 )
+      ac   = ta + hso4 - sit * hsi - ab - aw + ft * hf - pt * hpo4
+      ah2o = SQRT( ( tc - ac )**2 + 4. * ( ac * K2 / K1 ) * ( 2. * tc - ac ) )
+      ah2  = 0.5 * K1 / ac *( ( tc - ac ) + ah2o )
+      erel = ( ah2 - ah1 ) / ah2
+      if (abs( erel ).ge.eps) then
+        ah1 = ah2
+      else
+        exit iflag
+      endif
+    enddo iflag
+
+  end subroutine carchm_solve
+
+
+  subroutine carchm_solve_dicsat(saln,pco2,ta,sit,pt, &
+       Kh,K1,K2,Kb,Kw,Ks1,Kf,Ksi,K1p,K2p,K3p,         &
+       tc_sat,niter)
+
+    !**********************************************************************
+    ! Solve DICsat from TALK and pCO2.
+    !
+    ! J. Tjiputra,    *BCCR, Bergen*    25.01.17
+    !**********************************************************************
+
+    use mo_chemcon, only: bor1,bor2,salchl
+
+    real,    intent(in)  :: saln   ! salinity [psu].
+    real,    intent(in)  :: pco2   ! partial pressure of CO2 [ppm].
+    real,    intent(in)  :: ta     ! total alkalinity [eq/kg].
+    real,    intent(in)  :: sit    ! silicate concentration [mol/kg].
+    real,    intent(in)  :: pt     ! phosphate concentration [mol/kg].
+    real,    intent(in)  :: KH     ! equilibrium constant K0  = [H2CO3]/pCO2.
```

Comment:
`KH` -> `Kh` @JorgSchwinger : any reason, why the comment says `K0` and not `Kh`?

Reply:
Fixed for both the comment and the variable in the declaration.